framework/src/csifw/CSIParser.c

```diff
@@ -0,0 +1,117 @@
+/****************************************************************************
+ *
+ * Copyright 2024 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <stdio.h>
+#include "include/CSIParser.h"
+
+static void print_buf(const unsigned char* buf, int len)
+{
+	printf("RAW DATA %d", len);
+	unsigned long long *buff_tmp = (u64 *)buf;
+	int buff_len = (len / 8) + 1;
+	for (int i = 0; i < buff_len; i++) 
+		printf("[%02d]0x%016llx", i, buff_tmp[i]);
+}
+
+static float get_parsed_sub_carrier(const unsigned char* buf, int acc)
+{	
+	float Y;
+	if(acc)
+	{
+		Y = (buf[1]<<8) | buf[0];
+		if(Y>0x8000)
+			Y = (Y-0xffff-1)/(4096.0);
+		else
+			Y = Y/(4096.0);
+		return Y;
+
+	}
+	else
+	{
+		if(*buf>0x80)
+			Y = (*buf-0xff-1) / (16.0);
+		else
+			Y = *buf/(16.0);
+		return Y;
+	}
+}
+
+void logReceivedData(int csi_buff_len, unsigned char *csi_buff, int accuracy)
+{
+		#if 1
+		unsigned int timestamp;
+		unsigned int csi_seq_num;
+		/*LOGGING STARTS HERE*/
+		CSIFW_LOGI("================== LOGGING ==================");
+		print_buf(csi_buff, csi_buff_len);
+
+		//PRINTING PARSED DATA
+		CSIFW_LOGI("PRINTING PARSED DATA"); //csi_sequence, timestamp
+		printf("\n csi_buff_len (with header): %d, only csi len: 112",csi_buff_len);
+		printf("\nSub-carrier Count: 0x%x%x,", csi_buff[26], csi_buff[25]);
+		timestamp = (int)(csi_buff[18] << 24) | (int)(csi_buff[17] << 16) | (int)(csi_buff[16] << 8) | (int)csi_buff[15];
+		csi_seq_num = (int)(csi_buff[37] << 24) | (int)(csi_buff[36] << 16) | (int)(csi_buff[35] << 8) | (int)csi_buff[34];
+
+
+		CSIFW_LOGI("%d,%d,", csi_seq_num, timestamp); //csi_sequence, timestamp
+
+		printf("0x%x%x,", csi_buff[1], csi_buff[0]); //csi_signature
+		printf("0x%x,", csi_buff[2]); //hdr_len
+		CSIFW_LOGI("%x:%x:%x:%x:%x:%x,", csi_buff[8],csi_buff[7],csi_buff[6],csi_buff[5],csi_buff[4], csi_buff[3]); //mac_addr
+		CSIFW_LOGI("%x:%x:%x:%x:%x:%x,", csi_buff[14],csi_buff[13],csi_buff[12],csi_buff[11],csi_buff[10], csi_buff[9]); //trig_mac_addr
+		CSIFW_LOGI("%d,", ((int)csi_buff[31]/2 - 110)); //rssi
+
+		// CSIFW_LOGI("%s,",argv[1]); //real
+		CSIFW_LOGI("%d,", (int)csi_buff[19]); //channel, infered
```

Comment:
I don't think we are ok to print all raw data defaultly. I don't mean this highlighted lines only. There are many data printing in the framework.